File: android/app/src/main/java/com/cloudmusic/app/plugin/DeviceStoragePlugin.java
package com.cloudmusic.app.plugin;

import android.os.Environment;
import android.os.StatFs;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "DeviceStorage")
public class DeviceStoragePlugin extends Plugin {

    @PluginMethod
    public void getStorageInfo(PluginCall call) {
        try {
            StatFs stat = new StatFs(Environment.getExternalStorageDirectory().getPath());
            long bytesAvailable = stat.getBlockSizeLong() * stat.getAvailableBlocksLong();
            long bytesTotal = stat.getBlockSizeLong() * stat.getBlockCountLong();
            long bytesUsed = bytesTotal - bytesAvailable;

            JSObject ret = new JSObject();
            ret.put("total", bytesTotal);
            ret.put("used", bytesUsed);
            ret.put("free", bytesAvailable);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Could not get device storage", e);
        }
    }
}

File: android/app/src/main/java/com/cloudmusic/app/plugin/DeviceStoragePlugin.kt
package com.cloudmusic.app.plugin

import android.os.StatFs
import android.os.Environment
import com.getcapacitor.*
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "DeviceStorage")
class DeviceStoragePlugin : Plugin() {

    @PluginMethod
    fun getStorage(call: PluginCall) {
        try {
            val path = Environment.getDataDirectory()
            val stat = StatFs(path.path)

            val total = stat.blockSizeLong * stat.blockCountLong
            val free = stat.blockSizeLong * stat.availableBlocksLong
            val used = total - free

            val ret = JSObject()

            ret.put("total", total)
            ret.put("free", free)
            ret.put("used", used)

            call.resolve(ret)
        } catch (e: Exception) {
            call.reject("Could not get device storage", e)
        }
    }

    @PluginMethod
    fun getStorageInfo(call: PluginCall) {
        getStorage(call)
    }
}
